src/eslint/rules/__tests__/exaustive-inject.test.js

```diff
@@ -0,0 +1,166 @@
+import { RuleTester } from 'eslint';
+import rule from '../exhaustive-inject';
+
+RuleTester.setDefaultConfig({
+  parserOptions: {
+    ecmaVersion: 6,
+    sourceType: 'module',
+    ecmaFeatures: { jsx: true },
+  },
+});
+
+var ruleTester = new RuleTester();
+ruleTester.run('exhaustive-inject', rule, {
+  valid: [
+    // it should ignore built-in react components
+    `
+      import { Fragment } from 'react';
+      import { di } from 'react-magnetic-di';
+      import { useQuery } from 'react-apollo';
+
+      function MyComponent() {
+        di(useQuery);
+        useQuery();
+        return <Fragment />;
+      }
+    `,
+    // it should ignore not stateful react hooks
+    `
+      import { useMemo, useContext } from 'react';
+      import { di } from 'react-magnetic-di';
+
+      function MyComponent() {
+        di(useContext);
+        useMemo();
+        return useContext();
+      }
+    `,
+    {
+      // it should ignore components specified in options
+      code: `
+        import { Suspense, useState, useMemo } from 'react';
+        import { di } from 'react-magnetic-di';
+        import { Query } from 'react-apollo';
+  
+        function MyComponent() {
+          di(useState);
```

Comment:
should this inject `Suspense` as well?

Reply:
No as React built-ins are treated specially: by default we `di` only stateful hooks as other things (like `Suspense`, `Fragment`, `useMemo`, ... do not really make sense